app/src/main/java/org/mifos/mobile/ui/fragments/UpdatePasswordFragment.kt

```diff
@@ -63,6 +65,10 @@ class UpdatePasswordFragment : BaseFragment(), TextWatcher, OnFocusChangeListene
 
                 is RegistrationUiState.Error -> {
                     hideProgress()
+                    if (TextUtils.isEmpty(state.exception)) {
+                        showError(getString(R.string.not_found_error))
+                        return@observe
+                    }
```

Comment:
if I update the value of to `R.string.not_found_error` in case of an error then will I not be able to observe them later? why do we need to handle this explicitly in our ui ?

Reply:
I tried doing it that way but there is no context object available inside the ViewModel class. From the StackOverflow answer [here]( https://stackoverflow.com/questions/51451819/how-to-get-context-in-android-mvvm-viewmodel#:~:text=you%20can%20access%20the%20application,resources%2C%20preferences%2C%20etc..&text=Finally%20i%20got%20the%20easiest,context%20in%20viewModel%20using%20MVVM.) it is not a good practice to pass activity context in view model class. And since I do not have a context object, I can't add `getString(R.string.not_found_error)` in `RegistrationUiState.Error(exception)` inside ViewModel class, hence handling this here.

Reply:
um, I think there is some misunderstanding here. This block exclusively handles the case where we are checking 404 error and no error from API is coming. So need to show this string in UI. Otherwise I would have taken the error from json.

Reply:
I understand you are trying to handle 404 code, but what I was thinking was to change the type from String to Int in the RegistrationUiState sealed class. If we do say we can just update the value of live data to R.string.not_found_error and it would look Something like this : 
```
  _updatePasswordUiState.value = RegistrationUiState.Error(R.string.not_found_error)
```

In this way in our ui we will just need to show the error as  `showError(getString(state.exception))`. Thus we wont have to use TextUtils and JsonObject classes

Reply:
For that I will have to change the signature of RegistrationUiState.Error from string to int, then in my `extractErrorFromErrorBodyResponse` method how will I put the error which is coming in json as string? Or did you mean I put this generic error for any kind of error that's coming?

Reply:
Yes, put relevant errors in else and catch block. They should be enough for our use case. (We can get the code from response to display a more relevant messages in the future 🤔)

You won't be needing your function anymore if you do this

Reply:
made the changes.